server/aws-lsp-codewhisperer/src/language-server/agenticChat/tools/executeBash.ts

```diff
@@ -0,0 +1,349 @@
+                    firstChunk = false
+                },
+                onStderr: (chunk: string) => {
+                    ExecuteBash.handleChunk(firstStderrChunk ? '```console\n' + chunk : chunk, stderrBuffer, updates)
+                    firstStderrChunk = false
+                },
+            }
+
+            const childProcess = new processUtils.ChildProcess(
+                this.logger,
+                'bash',
+                ['-c', params.command],
+                childProcessOptions
+            )
+
+            try {
+                const result = await childProcess.run()
+                const exitStatus = result.exitCode ?? 0
+                const stdout = stdoutBuffer.join('\n')
+                const stderr = stderrBuffer.join('\n')
+                const [stdoutTrunc, stdoutSuffix] = ExecuteBash.truncateSafelyWithSuffix(
+                    stdout,
+                    maxBashToolResponseSize / 3
+                )
+                const [stderrTrunc, stderrSuffix] = ExecuteBash.truncateSafelyWithSuffix(
+                    stderr,
+                    maxBashToolResponseSize / 3
+                )
+
+                const outputJson = {
+                    exitStatus: exitStatus.toString(),
+                    stdout: stdoutTrunc + (stdoutSuffix ? ' ... truncated' : ''),
+                    stderr: stderrTrunc + (stderrSuffix ? ' ... truncated' : ''),
+                }
+
+                resolve({
+                    output: {
+                        kind: 'json',
+                        content: outputJson,
+                    },
+                })
+            } catch (err: any) {
+                this.logger.error(`Failed to execute bash command '${params.command}': ${err.message}`)
+                reject(new Error(`Failed to execute command: ${err.message}`))
+            }
+        })
+    }
+
+    private static handleChunk(chunk: string, buffer: string[], updates?: WritableStream) {
+        try {
+            const writer = updates?.getWriter()
+            void writer?.write(chunk)
+            const lines = chunk.split(/\r?\n/)
+            for (const line of lines) {
+                buffer.push(line)
+                if (buffer.length > lineCount) {
+                    buffer.shift()
+                }
+            }
+        } catch (error) {
+            // Log the error but don't let it crash the process
+            throw new Error('Error handling output chunk')
+        }
+    }
+
+    private static truncateSafelyWithSuffix(str: string, maxLength: number): [string, boolean] {
+        if (str.length > maxLength) {
+            return [str.substring(0, maxLength), true]
+        }
+        return [str, false]
+    }
+
+    private static async whichCommand(logger: Logging, cmd: string): Promise<string> {
+        const cp = new processUtils.ChildProcess(logger, 'which', [cmd], {
```

Comment:
The POSIX (i.e most-compatible) way is to use `command -v` (`sh -c 'command -v ...'`) or similar. Not sure if that matters, just FYI.

Reply:
That sounds like a better option. Additionally, I think these tests need to be refactored to avoid mocking the child process. This will allow us to confirm cross-platform support in CI.